content/en/docs/languages/ruby/basics.md

```diff
@@ -334,6 +334,33 @@ resps.each do |r|
 end
 ```
 
+Non-blocking usage of the RPC stream can be achieved with multiple threads and 
+the `return_op: true` flag. When passing the `return_op: true` flag, the 
+execution of the RPC is deferred and an `Operation` object is returned. The RPC 
+can then be executed in another thread by calling the operation `execute` 
+function. The main thread can utilize contextual methods and getters such as 
+`status`, `cancelled?`, and `cancel` to manage the connection. This can be 
```

Comment:
> This can be 
useful for managing persistent RPC sessions that may keep the connection open 
indefinitely.

While this is true, I'm not sure if we should include this text since there are other use cases for streaming RPCs (e.g. chunking very large messages), and the `Operation` view (e.g., the other accessors like `peer` and `metadata).

So in order don't want to it to sound like this is <i>the</i> intended use case for this feature, can we just leave this text out?

Reply:
I think it is helpful to give use case examples. When I first started diving into this, I was constantly questioning myself - is this an appropriate use case for using this code? The documentation states `Note that the gRPC Ruby only
provides  *blocking/synchronous* versions of each method`. It made me ask, "Is this implying that non-blocking persistent RPC sessions are not supported in the ruby library?"

Utilizing the wording "can be useful" implies (at least to me) that this is not the sole use case. The developer can see the type of scenarios this should support and branch out from there. I will expand this to make this point even more clear.